File: ios/Sonet/NativeApp/Views/Home/HomeTabView.swift
import SwiftUI

struct HomeTabView: View {
    @EnvironmentObject var navigationManager: NavigationManager
    @EnvironmentObject var themeManager: ThemeManager
    @StateObject private var homeViewModel = HomeViewModel()
    
    var body: some View {
        NavigationView {
            VStack(spacing: 0) {
                // Custom Navigation Bar
                HomeNavigationBar(
                    selectedFeed: $homeViewModel.selectedFeed,
                    onFeedChanged: { feed in
                        homeViewModel.selectFeed(feed)
                    }
                )
                
                // Feed Content
                if homeViewModel.isLoading {
                    LoadingView()
                } else if let error = homeViewModel.error {
                    ErrorView(error: error) {
                        homeViewModel.refreshFeed()
                    }
                } else {
                    FeedView(
                        feed: homeViewModel.selectedFeed,
                        notes: homeViewModel.notes,
                        onRefresh: {
                            homeViewModel.refreshFeed()
                        },
                        onLoadMore: {
                            homeViewModel.loadMoreNotes()
                        }
                    )
                }
            }
            .navigationBarHidden(true)
        }
        .navigationViewStyle(StackNavigationViewStyle())
        .onAppear {
            homeViewModel.loadFeed()
        }
    }
}

// MARK: - Home Navigation Bar
struct HomeNavigationBar: View {
    @Binding var selectedFeed: FeedType
    let onFeedChanged: (FeedType) -> Void
    
    var body: some View {
        VStack(spacing: 0) {
            // Main Header
            HStack {
                Text("Home")
                    .font(.largeTitle)
                    .fontWeight(.bold)
                
                Spacer()
                
                Button(action: { /* Open composer */ }) { IconView(AppIcons.add, size: 22, color: .blue) }
            }
            .padding(.horizontal, 16)
            .padding(.top, 8)
            
            // Feed Type Selector
            ScrollView(.horizontal, showsIndicators: false) {
                HStack(spacing: 16) {
                    ForEach(FeedType.allCases, id: \.self) { feedType in
                        FeedTypeButton(
                            feedType: feedType,
                            isSelected: selectedFeed == feedType
                        ) {
                            selectedFeed = feedType
                            onFeedChanged(feedType)
                        }
                    }
                }
                .padding(.horizontal, 16)
            }
            .padding(.vertical, 8)
            
            Divider()
        }
        .background(Color(.systemBackground))
    }
}

// MARK: - Feed Type Button
struct FeedTypeButton: View {
    let feedType: FeedType
    let isSelected: Bool
    let action: () -> Void
    
    var body: some View {
        Button(action: action) {
            Text(feedType.displayName)
                .font(.subheadline)
                .fontWeight(isSelected ? .semibold : .medium)
                .foregroundColor(isSelected ? .white : .primary)
                .padding(.horizontal, 16)
                .padding(.vertical, 8)
                .background(
                    RoundedRectangle(cornerRadius: 20)
                        .fill(isSelected ? Color.blue : Color.clear)
                )
                .overlay(
                    RoundedRectangle(cornerRadius: 20)
                        .stroke(isSelected ? Color.clear : Color.gray.opacity(0.3), lineWidth: 1)
                )
        }
        .buttonStyle(PlainButtonStyle())
    }
}

// MARK: - Feed View
struct FeedView: View {
    let feed: FeedType
    let notes: [SonetNote]
    let onRefresh: () -> Void
    let onLoadMore: () -> Void
    
    var body: some View {
        ScrollView {
            LazyVStack(spacing: 16) {
                ForEach(notes) { note in
                    NoteCard(note: note)
                        .padding(.horizontal, 16)
                }
                
                // Load More Button
                if !notes.isEmpty {
                    Button(action: onLoadMore) {
                        Text("Load More")
                            .font(.subheadline)
                            .foregroundColor(.blue)
                            .padding(.vertical, 12)
                    }
                    .padding(.horizontal, 16)
                }
            }
            .padding(.vertical, 16)
        }
        .refreshable {
            onRefresh()
        }
    }
}

// MARK: - Note Card
struct NoteCard: View {
    let note: SonetNote
    @State private var isLiked = false
    @State private var isReposted = false
    
    var body: some View {
        VStack(alignment: .leading, spacing: 12) {
            // User Header
            HStack {
                AsyncImage(url: note.author.avatarURL) { image in
                    image
                        .resizable()
                        .aspectRatio(contentMode: .fill)
                } placeholder: {
                    IconView(AppIcons.personCircle, size: 16, color: .gray)
                }
                .frame(width: 40, height: 40)
                .clipShape(Circle())
                
                VStack(alignment: .leading, spacing: 2) {
                    HStack {
                        Text(note.author.displayName)
                            .font(.subheadline)
                            .fontWeight(.semibold)
                        
                        if note.author.isVerified { IconView(AppIcons.verified, size: 12, color: .blue) }
                    }
                    
                    Text(note.author.handle)
                        .font(.caption)
                        .foregroundColor(.secondary)
                }
                
                Spacer()
                
                Text(note.createdAt.timeAgoDisplay())
                    .font(.caption)
                    .foregroundColor(.secondary)
            }
            
            // Note Content
            if !note.text.isEmpty {
                Text(note.text)
                    .font(.body)
                    .multilineTextAlignment(.leading)
            }
            
            // Media Content
            if let media = note.media, !media.isEmpty {
                MediaGridView(media: media)
            }
            
            // Action Buttons
            HStack(spacing: 24) {
                // Reply Button
                Button(action: {
                    // Handle reply
                }) {
                    HStack(spacing: 4) {
                        Image(systemName: "bubble.left")
                        Text("\(note.replyCount)")
                    }
                    .font(.caption)
                    .foregroundColor(.secondary)
                }
                
                // Repost Button
                Button(action: {
                    isReposted.toggle()
                }) {
                    HStack(spacing: 4) {
                        Image(systemName: isReposted ? "arrow.2.squarepath.fill" : "arrow.2.squarepath")
                        Text("\(note.repostCount)")
                    }
                    .font(.caption)
                    .foregroundColor(isReposted ? .green : .secondary)
                }
                
                // Like Button
                Button(action: {
                    isLiked.toggle()
                }) {
                    HStack(spacing: 4) {
                        Image(systemName: isLiked ? "heart.fill" : "heart")
                        Text("\(note.likeCount)")
                    }
                    .font(.caption)
                    .foregroundColor(isLiked ? .red : .secondary)
                }
                
                // Share Button
                Button(action: {
                    // Handle share
                }) {
                    Image(systemName: "square.and.arrow.up")
                        .font(.caption)
                        .foregroundColor(.secondary)
                }
                
                Spacer()
            }
        }
        .padding(16)
        .background(Color(.systemBackground))
        .cornerRadius(12)
        .shadow(color: .black.opacity(0.05), radius: 2, x: 0, y: 1)
    }
}

// MARK: - Media Grid View
struct MediaGridView: View {
    let media: [MediaItem]
    
    var body: some View {
        let columns = media.count == 1 ? 1 : 2
        
        LazyVGrid(columns: Array(repeating: GridItem(.flexible(), spacing: 4), count: columns), spacing: 4) {
            ForEach(media) { item in
                AsyncImage(url: item.url) { image in
                    image
                        .resizable()
                        .aspectRatio(contentMode: .fill)
                } placeholder: {
                    Rectangle()
                        .fill(Color.gray.opacity(0.3))
                        .overlay(
                            ProgressView()
                        )
                }
                .frame(height: columns == 1 ? 200 : 100)
                .clipped()
                .cornerRadius(8)
            }
        }
    }
}

// MARK: - Loading View
struct LoadingView: View {
    var body: some View {
        VStack(spacing: 16) {
            ProgressView()
                .scaleEffect(1.2)
            
            Text("Loading feed...")
                .font(.subheadline)
                .foregroundColor(.secondary)
        }
        .frame(maxWidth: .infinity, maxHeight: .infinity)
    }
}

// MARK: - Error View
struct ErrorView: View {
    let error: Error
    let retryAction: () -> Void
    
    var body: some View {
        VStack(spacing: 16) {
            Image(systemName: "exclamationmark.triangle")
                .font(.system(size: 48))
                .foregroundColor(.orange)
            
            Text("Something went wrong")
                .font(.headline)
            
            Text(error.localizedDescription)
                .font(.subheadline)
                .foregroundColor(.secondary)
                .multilineTextAlignment(.center)
                .padding(.horizontal, 32)
            
            Button("Try Again") {
                retryAction()
            }
            .buttonStyle(.borderedProminent)
        }
        .frame(maxWidth: .infinity, maxHeight: .infinity)
        .padding()
    }
}

// MARK: - Preview
struct HomeTabView_Previews: PreviewProvider {
    static var previews: some View {
        HomeTabView()
            .environmentObject(NavigationManager())
            .environmentObject(ThemeManager())
    }
}

File: android/app/src/main/java/xyz/sonet/app/views/home/HomeTabView.kt
package xyz.sonet.app.views.home

import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.*
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import xyz.sonet.app.viewmodels.SessionViewModel
import xyz.sonet.app.viewmodels.ThemeViewModel
import xyz.sonet.app.viewmodels.HomeViewModel

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun HomeTabView(
    sessionViewModel: SessionViewModel,
    themeViewModel: ThemeViewModel
) {
    val homeViewModel = androidx.lifecycle.viewmodel.compose.viewModel<HomeViewModel>()
    
    Column(
        modifier = Modifier.fillMaxSize()
    ) {
        // Custom Navigation Bar
        HomeNavigationBar(
            selectedFeed = homeViewModel.selectedFeed,
            onFeedChanged = { feed ->
                homeViewModel.selectFeed(feed)
            }
        )
        
        // Feed Content
        when {
            homeViewModel.isLoading -> {
                LoadingView()
            }
            homeViewModel.error != null -> {
                ErrorView(
                    error = homeViewModel.error!!,
                    onRetry = {
                        homeViewModel.refreshFeed()
                    }
                )
            }
            else -> {
                FeedView(
                    feed = homeViewModel.selectedFeed,
                    notes = homeViewModel.notes,
                    onRefresh = {
                        homeViewModel.refreshFeed()
                    },
                    onLoadMore = {
                        homeViewModel.loadMoreNotes()
                    }
                )
            }
        }
    }
    
    LaunchedEffect(Unit) {
        homeViewModel.loadFeed()
    }
}

@Composable
fun HomeNavigationBar(
    selectedFeed: FeedType,
    onFeedChanged: (FeedType) -> Unit
) {
    Column(
        modifier = Modifier.fillMaxWidth()
    ) {
        // Main Header
        Row(
            modifier = Modifier
                .fillMaxWidth()
                .padding(horizontal = 16.dp, vertical = 8.dp),
            verticalAlignment = Alignment.CenterVertically
        ) {
            Text(
                text = "Home",
                style = MaterialTheme.typography.headlineLarge,
                fontWeight = FontWeight.Bold
            )
            
            Spacer(modifier = Modifier.weight(1f))
            
            IconButton(
                onClick = { /* Open composer */ }
            ) {
                Icon(
                    imageVector = Icons.Default.Add,
                    contentDescription = "Compose",
                    tint = MaterialTheme.colorScheme.primary
                )
            }
        }
        
        // Feed Type Selector
        LazyRow(
            contentPadding = PaddingValues(horizontal = 16.dp, vertical = 8.dp),
            horizontalArrangement = Arrangement.spacedBy(16.dp)
        ) {
            items(FeedType.values()) { feedType ->
                FeedTypeButton(
                    feedType = feedType,
                    isSelected = selectedFeed == feedType,
                    onClick = {
                        onFeedChanged(feedType)
                    }
                )
            }
        }
        
        Divider()
    }
}

@Composable
fun FeedTypeButton(
    feedType: FeedType,
    isSelected: Boolean,
    onClick: () -> Unit
) {
    Button(
        onClick = onClick,
        colors = ButtonDefaults.buttonColors(
            containerColor = if (isSelected) MaterialTheme.colorScheme.primary else MaterialTheme.colorScheme.surface,
            contentColor = if (isSelected) MaterialTheme.colorScheme.onPrimary else MaterialTheme.colorScheme.onSurface
        ),
        shape = MaterialTheme.shapes.medium
    ) {
        Text(
            text = feedType.displayName,
            style = MaterialTheme.typography.bodyMedium,
            fontWeight = if (isSelected) FontWeight.Semibold else FontWeight.Medium
        )
    }
}

@Composable
fun FeedView(
    feed: FeedType,
    notes: List<SonetNote>,
    onRefresh: () -> Unit,
    onLoadMore: () -> Unit
) {
    LazyColumn(
        contentPadding = PaddingValues(16.dp),
        verticalArrangement = Arrangement.spacedBy(16.dp)
    ) {
        items(notes) { note ->
            NoteCard(note = note)
        }
        
        // Load More Button
        if (notes.isNotEmpty()) {
            item {
                Button(
                    onClick = onLoadMore,
                    modifier = Modifier.fillMaxWidth()
                ) {
                    Text("Load More")
                }
            }
        }
    }
}

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun NoteCard(note: SonetNote) {
    var isLiked by remember { mutableStateOf(false) }
    var isReposted by remember { mutableStateOf(false) }
    
    Card(
        modifier = Modifier.fillMaxWidth(),
        elevation = CardDefaults.cardElevation(defaultElevation = 2.dp)
    ) {
        Column(
            modifier = Modifier.padding(16.dp),
            verticalArrangement = Arrangement.spacedBy(12.dp)
        ) {
            // User Header
            Row(
                verticalAlignment = Alignment.CenterVertically
            ) {
                // Avatar placeholder
                Surface(
                    modifier = Modifier.size(40.dp),
                    shape = MaterialTheme.shapes.circular,
                    color = MaterialTheme.colorScheme.surfaceVariant
                ) {
                    Icon(
                        imageVector = Icons.Default.Person,
                        contentDescription = null,
                        modifier = Modifier.padding(8.dp)
                    )
                }
                
                Spacer(modifier = Modifier.width(12.dp))
                
                Column {
                    Row(
                        verticalAlignment = Alignment.CenterVertically
                    ) {
                        Text(
                            text = note.author.displayName,
                            style = MaterialTheme.typography.bodyMedium,
                            fontWeight = FontWeight.Semibold
                        )
                        
                        if (note.author.isVerified) {
                            Spacer(modifier = Modifier.width(4.dp))
                            Icon(
                                imageVector = Icons.Default.Verified,
                                contentDescription = "Verified",
                                tint = MaterialTheme.colorScheme.primary,
                                modifier = Modifier.size(16.dp)
                            )
                        }
                    }
                    
                    Text(
                        text = note.author.handle,
                        style = MaterialTheme.typography.bodySmall,
                        color = MaterialTheme.colorScheme.onSurfaceVariant
                    )
                }
                
                Spacer(modifier = Modifier.weight(1f))
                
                Text(
                    text = note.timeAgo,
                    style = MaterialTheme.typography.bodySmall,
                    color = MaterialTheme.colorScheme.onSurfaceVariant
                )
            }
            
            // Note Content
            if (note.text.isNotEmpty()) {
                Text(
                    text = note.text,
                    style = MaterialTheme.typography.bodyLarge
                )
            }
            
            // Media Content placeholder
            if (note.media != null && note.media.isNotEmpty()) {
                Surface(
                    modifier = Modifier
                        .fillMaxWidth()
                        .height(200.dp),
                    shape = MaterialTheme.shapes.medium,
                    color = MaterialTheme.colorScheme.surfaceVariant
                ) {
                    Box(
                        modifier = Modifier.fillMaxSize(),
                        contentAlignment = Alignment.Center
                    ) {
                        Icon(
                            imageVector = Icons.Default.Image,
                            contentDescription = "Media",
                            modifier = Modifier.size(48.dp),
                            tint = MaterialTheme.colorScheme.onSurfaceVariant
                        )
                    }
                }
            }
            
            // Action Buttons
            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.SpaceBetween
            ) {
                // Reply Button
                IconButton(onClick = { /* Handle reply */ }) {
                    Row(verticalAlignment = Alignment.CenterVertically) {
                        Icon(
                            imageVector = Icons.Default.ChatBubbleOutline,
                            contentDescription = "Reply"
                        )
                        Spacer(modifier = Modifier.width(4.dp))
                        Text("${note.replyCount}")
                    }
                }
                
                // Repost Button
                IconButton(
                    onClick = { isReposted = !isReposted }
                ) {
                    Row(verticalAlignment = Alignment.CenterVertically) {
                        Icon(
                            imageVector = if (isReposted) Icons.Default.Reply else Icons.Default.Reply,
                            contentDescription = "Repost",
                            tint = if (isReposted) MaterialTheme.colorScheme.primary else MaterialTheme.colorScheme.onSurfaceVariant
                        )
                        Spacer(modifier = Modifier.width(4.dp))
                        Text("${note.repostCount}")
                    }
                }
                
                // Like Button
                IconButton(
                    onClick = { isLiked = !isLiked }
                ) {
                    Row(verticalAlignment = Alignment.CenterVertically) {
                        Icon(
                            imageVector = if (isLiked) Icons.Default.Favorite else Icons.Default.FavoriteBorder,
                            contentDescription = "Like",
                            tint = if (isLiked) MaterialTheme.colorScheme.error else MaterialTheme.colorScheme.onSurfaceVariant
                        )
                        Spacer(modifier = Modifier.width(4.dp))
                        Text("${note.likeCount}")
                    }
                }
                
                // Share Button
                IconButton(onClick = { /* Handle share */ }) {
                    Icon(
                        imageVector = Icons.Default.Share,
                        contentDescription = "Share"
                    )
                }
            }
        }
    }
}

@Composable
fun LoadingView() {
    Box(
        modifier = Modifier.fillMaxSize(),
        contentAlignment = Alignment.Center
    ) {
        Column(
            horizontalAlignment = Alignment.CenterHorizontally,
            verticalArrangement = Arrangement.spacedBy(16.dp)
        ) {
            CircularProgressIndicator(
                modifier = Modifier.size(48.dp)
            )
            
            Text(
                text = "Loading feed...",
                style = MaterialTheme.typography.bodyMedium,
                color = MaterialTheme.colorScheme.onSurfaceVariant
            )
        }
    }
}

@Composable
fun ErrorView(
    error: Throwable,
    onRetry: () -> Unit
) {
    Box(
        modifier = Modifier.fillMaxSize(),
        contentAlignment = Alignment.Center
    ) {
        Column(
            horizontalAlignment = Alignment.CenterHorizontally,
            verticalArrangement = Arrangement.spacedBy(16.dp)
        ) {
            Icon(
                imageVector = Icons.Default.Error,
                contentDescription = null,
                modifier = Modifier.size(48.dp),
                tint = MaterialTheme.colorScheme.error
            )
            
            Text(
                text = "Something went wrong",
                style = MaterialTheme.typography.headlineSmall
            )
            
            Text(
                text = error.message ?: "An unknown error occurred",
                style = MaterialTheme.typography.bodyMedium,
                color = MaterialTheme.colorScheme.onSurfaceVariant,
                textAlign = TextAlign.Center
            )
            
            Button(onClick = onRetry) {
                Text("Try Again")
            }
        }
    }
}

// Data models (these would typically be in separate files)
enum class FeedType(val displayName: String) {
    FOLLOWING("Following"),
    FOR_YOU("For You"),
    TRENDING("Trending"),
    LATEST("Latest")
}

data class SonetNote(
    val id: String,
    val text: String,
    val author: SonetUser,
    val createdAt: Long,
    val likeCount: Int,
    val repostCount: Int,
    val replyCount: Int,
    val media: List<MediaItem>?,
    var isLiked: Boolean,
    var isReposted: Boolean
) {
    val timeAgo: String
        get() = "2h ago" // Simplified for now
}

data class SonetUser(
    val id: String,
    val username: String,
    val displayName: String,
    val avatarUrl: String?,
    val isVerified: Boolean,
    val createdAt: Long
) {
    val handle: String
        get() = "@$username"
}

data class MediaItem(
    val id: String,
    val url: String,
    val type: MediaType,
    val altText: String?
)

enum class MediaType {
    IMAGE, VIDEO, GIF
}